walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/identifier/resolvers/Resolver.kt

```diff
@@ -18,13 +16,12 @@ internal class Resolver @Inject constructor(
     private val identifierRepository: IdentifierRepository
 ) {
     suspend fun resolve(identifier: String): Result<IdentifierDocument> {
-        return runResultTry {
-            when (val id = identifierRepository.resolveIdentifier(baseUrl, identifier)) {
-                is Result.Success -> {
-                    Result.Success(id.payload.didDocument)
-                }
-                is Result.Failure -> Result.Failure(ResolverException("Unable to resolve identifier $identifier", id.payload))
+        return identifierRepository.resolveIdentifier(baseUrl, identifier)
+            .map {
+                it.didDocument
+            }
+            .onFailure {
+                return Result.failure(ResolverException("Unable to resolve identifier $identifier"))
```

Comment:
Can we populate the cause here along with the message for debugging the specific failure cause?